src/rgw/rgw_sal_motr.cc

```diff
@@ -2192,29 +2195,63 @@ int MotrObject::read_mobj(const DoutPrefixProvider* dpp, int64_t off, int64_t en
 
 int MotrObject::get_bucket_dir_ent(const DoutPrefixProvider *dpp, rgw_bucket_dir_entry& ent)
 {
-  int rc = 0;
+  int rc = 0, ret_code = -ENOENT;
   string tenant_bkt_name = get_bucket_name(this->get_bucket()->get_tenant(), this->get_bucket()->get_name());
   string bucket_index_iname = "motr.rgw.bucket.index." + tenant_bkt_name;
-  int max = 1000;
+  int max = 2;
   vector<string> keys(max);
   vector<bufferlist> vals(max);
   bufferlist bl;
   bufferlist::const_iterator iter;
 
-  if (this->get_bucket()->get_info().versioning_status() == BUCKET_VERSIONED ||
-      this->get_bucket()->get_info().versioning_status() == BUCKET_SUSPENDED) {
+  // Check entry in the cache
+  if (this->store->get_obj_meta_cache()->get(dpp, this->get_name(), bl) == 0) {
```

Comment:
What's the purpose of this if block here. Are we try to fetch the latest version? is applicable for all the bucket version state?
I think need to be very precise on the use case. After making the version related changes we can not be very generic while doing the lookup for the keys in the index or cache.

Reply:
We have refactored the code here.
done